Loop/Managers/DeviceDataManager.swift

```diff
@@ -425,19 +479,25 @@ final class DeviceDataManager {
 
         return Manager.init(rawState: rawState) as? PumpManagerUI
     }
+    
+    private func checkPumpDataAndLoop() {
+        self.log.default("Asserting current pump data")
+        guard let pumpManager = pumpManager else {
+            self.loopManager.loop()
+            return
+        }
+
+        pumpManager.ensureCurrentPumpData() { (lastSync) in
+            self.loopManager.loop()
+        }
```

Comment:
previously, there was a time check on lastSync before looping. Is that not needed anymore?

Reply:
LDM will check the pump recency and fail if it's not recent enough. This allows us to record in dosing decision store that a loop was attempted and failed due to old pump data, which allows remote clients, like Nightscout (and eventually Tidepool backend) to report the reason for loop failure to remote caregivers.